src/libcore/iter/traits/iterator.rs

```diff
@@ -2038,12 +2034,8 @@ pub trait Iterator {
     #[stable(feature = "rust1", since = "1.0.0")]
     fn min(self) -> Option<Self::Item> where Self: Sized, Self::Item: Ord
     {
-        select_fold1(self,
-                     |_| (),
-                     // only switch to y if it is strictly smaller, to
-                     // preserve stability.
-                     |_, x, _, y| *x > *y)
-            .map(|(_, x)| x)
+        // only switch to y if it is strictly smaller, to preserve stability.
```

Comment:
Can we just have this delegate to min_by?

Reply:
We indeed can, I only considered forwarding to `min_by_key` 🙂 Thanks!